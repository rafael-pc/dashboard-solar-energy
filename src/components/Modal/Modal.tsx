import { FC } from "react";
import { Modal as Modals, Box, Collapse, IconButton } from "@mui/material";
import CloseIcon from "@mui/icons-material/Close";
import Alert from "@mui/material/Alert";

interface IModalProps {
  isOpen: boolean;
  message: string;
  isSuccessful: boolean;
  setOpen: (isOpen: boolean) => void;
  handleClose: () => void;
}

const Modal: FC<IModalProps> = ({isOpen, message, isSuccessful, setOpen, handleClose}) => {
  const BoxStyled = {
    position: "absolute",
    top: "8%",
    left: "50%",
    transform: "translate(-50%, -50%)",
    width: 350,
  };

  return (
    <Modals
      open={isOpen}
      onClose={handleClose}
      aria-labelledby="modal-modal-title"
      aria-describedby="modal-modal-description"
    >
      <Box sx={BoxStyled}>
        <Collapse in={isOpen}>
          {isSuccessful ? (
            <Alert
              severity="success"
              action={
                <IconButton
                  aria-label="close"
                  color="inherit"
                  size="small"
                  onClick={handleClose}
                >
                  <CloseIcon fontSize="inherit" />
                </IconButton>
              }
            >
              {message}
            </Alert>
          ) : (
            <Alert
              severity="error"
              action={
                <IconButton
                  aria-label="close"
                  color="inherit"
                  size="small"
                  onClick={() => {
                    setOpen(false);
                  }}
                >
                  <CloseIcon fontSize="inherit" />
                </IconButton>
              }
            >
              {message}
            </Alert>
          )}
        </Collapse>
      </Box>
    </Modals>
  );
};

export default Modal;
