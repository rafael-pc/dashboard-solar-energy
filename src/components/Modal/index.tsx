export { default } from "./Modal";
//export { default as ModalRegisterMonth} from "./ModalRegisterMonth";
