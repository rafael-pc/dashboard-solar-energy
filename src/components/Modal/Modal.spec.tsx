import { render, screen } from '@testing-library/react';
import Modal from './Modal';
import "@testing-library/jest-dom";

describe('Modal', () => {
  it('renders message and close button when open', () => {
    const handleClose = jest.fn();
    const { asFragment } = render(
      <Modal
        isOpen={true}
        message="Success!"
        isSuccessful={true}
        setOpen={() => {""}}
        handleClose={handleClose}
      />
    );
    expect(asFragment()).toMatchSnapshot();

    expect(screen.getByText('Success!')).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'close' })).toBeInTheDocument();
  });

  it('does not render message and close button when closed', () => {
    const handleClose = jest.fn();
    render(
      <Modal
        isOpen={false}
        message="Success!"
        isSuccessful={true}
        setOpen={() => {""}}
        handleClose={handleClose}
      />
    );
    expect(screen.queryByText('Success!')).not.toBeInTheDocument();
    expect(screen.queryByRole('button', { name: 'close' })).not.toBeInTheDocument();
  });
});
