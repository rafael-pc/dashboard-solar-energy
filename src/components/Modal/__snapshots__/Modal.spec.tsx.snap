// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders message and close button when open 1`] = `<DocumentFragment />`;
